types.go

```diff
@@ -22,37 +22,42 @@ import (
 )
 
 type healthDependency struct {
-	Name              string    `json:"name"`
-	URL               string    `json:"url"`
-	Healthy           bool      `json:"healthy"`
-	HardDependency    bool      `json:"hardDependency"`
-	LastKnownGoodCall time.Time `json:"lastKnownGoodCall"`
-	LastCall          time.Time `json:"lastCall"`
-	LastError         LastError `json:"lastError"`
+	Name              string     `json:"name"`
+	URL               string     `json:"url"`
+	Healthy           bool       `json:"healthy"`
+	HardDependency    bool       `json:"hardDependency"`
+	LastKnownGoodCall *time.Time `json:"lastKnownGoodCall,omitempty"`
+	LastCall          *time.Time `json:"lastCall,omitempty"`
+	LastError         *LastError `json:"lastError,omitempty"`
 	checkFunc         CheckFunc
 }
 
 type LastError struct {
-	timestamp time.Time `json:"timestamp"`
-	message   string    `json:"message"`
+	Timestamp *time.Time `json:"timestamp"`
+	Message   string     `json:"message"`
 }
 
 func (h *healthDependency) check() {
 	if h.checkFunc == nil {
 		return
 	}
 
-	h.LastCall = time.Now()
+	now := time.Now()
+	h.LastCall = &now
 	err := h.checkFunc()
 	if err != nil {
 		logrus.Error(err)
-		h.LastError.message = err.Error()
-		h.LastError.timestamp = h.LastCall
+		if h.LastError == nil {
+			h.LastError = &LastError{}
+		}
+		h.LastError.Message = err.Error()
+		h.LastError.Timestamp = h.LastCall
 		h.Healthy = false
 		return
 	}
 	h.Healthy = true
-	h.LastKnownGoodCall = time.Now()
+	h.LastKnownGoodCall = &now
+	h.LastError = nil
```

Comment:
@thoriqsatriya do we want to retain last error even after the check is healthy?

Reply:
yes, it would help us check the last error (if any)

Reply:
yeah that would be helpful. I have removed the line. Thanks